### PR TITLE
refactor(reports): fix literal-string violations in xl/xlt calls

### PR DESCRIPTION
**Pull request by kojiromike**
## Summary

Part 6 of a series fixing ~225 PHPStan literal-string violations in `xl()`/`xlt()`/`xla()` calls.

Fix PHPStan literal-string violations in reports and billing files:

- **patient_list.php**: drop spurious `xl`/`xlt` on street/city/state (addresses are not translatable); narrow with `is_string()` before echoing.
- **message_list.php**: switch `xl()` on DB labels to `xl_list_label()` and narrow `$msg_type`/`$msg_status` with `is_string()`.
- **ippf_statistics.php**: switch `xl()` on `lo_title` to `xl_list_label()` with `is_string()` narrowing.
- **receipts_by_method_report.php**: drop spurious `xl()` on insurance company name (not translatable); narrow with `is_string()`.
- **billing_tracker_ajax.php**: switch `xl()` on claim status to `xl_list_label()` with `is_string()` narrowing.
- **FeeSheet.class.php**: inline `'COPAY'` literal since `$codetype` is always `'COPAY'` inside the `if` block.
- **payment_master.inc.php**: add native `string` types to `generate_list_payment_category()`; switch `xlt($empty_name)` to `text(xl_list_label($empty_name))`; narrow `$lrow['title']` with `is_string()`.

PHPStan baseline counts only decrease — no new entries added.

## Test plan

- [x] `composer phpstan` reports no errors
- [x] `composer phpstan-baseline` shows only decreases for touched files
- [x] Pre-commit hooks pass (phpcs, phpstan, rector, etc.)

### CODE REVIEW

File: library/ajax/billing_tracker_ajax.php [hunk 1]
```diff
@@ -39,7 +39,7 @@
     $element->x12_partner_id = text($claim_file['x12_partner_id']);
     $element->x12_partner_name = text($claim_file['name']);
     $element->x12_filename = text($claim_file['x12_filename']);
-    $element->status = xl($claim_file['status']);
+    $element->status = xl_list_label(is_string($claim_file['status'] ?? null) ? $claim_file['status'] : '');
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply library/ajax/billing_tracker_ajax.php:42]: Good point about the JS comparing translated strings to internal status values — but this is a pre-existing bug. The line was previously `$element->status = xl($claim_file['status'])`, which already translated the value before sending it to the JS. My change only swapped `xl()` for `xl_list_label()` (which is the equivalent literal-string-safe wrapper) and added the `is_string()` narrowing required by PHPStan. Splitting `status` from `status_label` is the right long-term fix, but it's a behavior change unrelated to the literal-string refactor. I'd rather file a follow-up issue and keep this PR scoped. Resolving as out of scope.

File: interface/billing/payment_master.inc.php [hunk 2]
```diff
@@ -15,25 +15,25 @@
...
 {
     $s = '';
     $s .= "<select name='" . attr($tag_name) . "' id='" . attr($tag_name) . "'";
-    if ($class) {
+    if ($class !== '') {
         $s .= " class='" . attr($class) . "'";
     }
-    if ($onchange) {
+    if ($onchange !== '') {
         $s .= " onchange='" . $onchange . "'"; //Need to html escape $onchange prior to the generate_list_payment_category function call
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply interface/billing/payment_master.inc.php:26]: The unescaped `$onchange` predates this PR — note the existing inline comment "Need to html escape $onchange prior to the generate_list_payment_category function call" sitting right above the line, which encodes the contract that callers escape. My diff only added native parameter types and return type and tightened the `if ($onchange)` checks; I didn't touch the concatenation. Hardening this helper to escape internally is a worthwhile fix but it's a behavior change for every caller and out of scope here. Filing a follow-up makes more sense than mixing it into a literal-string PR. Resolving as out of scope.

File: interface/reports/receipts_by_method_report.php [hunk 3]
```diff
@@ -668,7 +668,7 @@ function editInvoice(e, id) {
                     }
                     if (!empty($insurance_id['provider'])) {
                         $insurance_company = (new InsuranceCompanyService())->getOneById($insurance_id['provider']) ?? '';
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply interface/reports/receipts_by_method_report.php]: Good catch — confirmed `getOneById()` delegates to `sqlQuery()` which returns `array|false|null`, so `?? ''` let `false` through. Fixed in `1123dad5` using `?: []` as suggested. Then the existing `is_string($insurance_company['name'] ?? null)` narrowing is safe because `$insurance_company` is guaranteed to be an array.

File: interface/reports/receipts_by_method_report.php [hunk 4]
```diff
@@ -667,8 +667,11 @@ function editInvoice(e, id) {
                         $rowmethod = xl('Unnamed insurance company');
                     }
                     if (!empty($insurance_id['provider'])) {
-                        $insurance_company = (new InsuranceCompanyService())->getOneById($insurance_id['provider']) ?? '';
-                        $rowmethod = xl($insurance_company['name']);
+                        // getOneById delegates to sqlQuery which can return
+                        // array|false|null. Normalize to [] so the ['name']
+                        // lookup below can't trip on a non-array value.
+                        $insurance_company = (new InsuranceCompanyService())->getOneById($insurance_id['provider']) ?: [];
+                        $rowmethod = is_string($insurance_company['name'] ?? null) ? $insurance_company['name'] : '';
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply interface/reports/receipts_by_method_report.php]: Good catch — the empty-string fallback was silently bucketing receipts with missing insurance company names as "Patient"/"Unknown" downstream. Fixed in `9511c623` using your suggested structure: check `is_string` + non-empty after trim, fall back to `xl('Unnamed insurance company')` to match the label the missing-provider branch below uses.

File: interface/billing/payment_master.inc.php [hunk 5]
```diff
@@ -43,9 +43,10 @@ function generate_list_payment_category($tag_name, $list_id, $currvalue, $title,
         if ($PaymentType == 'patient' && $lrow['option_id'] == 'insurance_payment') {
             $s .= " style='background-color: var(--light)' ";
         }
-        $s .= ">" . text(xl_list_label($lrow['title'])) . "</option>\n";
+        $title_value = is_string($lrow['title'] ?? null) ? $lrow['title'] : '';
+        $s .= ">" . text(xl_list_label($title_value)) . "</option>\n";
     }
-    if (!$got_selected && strlen((string) $currvalue) > 0) {
+    if (!$got_selected && strlen($currvalue) > 0) {
         $currescaped = text($currvalue);
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply interface/billing/payment_master.inc.php]: Removed in `a728b007` — `$currescaped` was indeed dead. The `text($currvalue)` call inline below already handles the escaping for the option label.

File: interface/reports/patient_list.php [hunk 6]
```diff
@@ -280,15 +280,19 @@
             $age = intval($ageInMonths / 12);
         }
 
+        $street = is_string($row['street'] ?? null) ? $row['street'] : '';
+        $city = is_string($row['city'] ?? null) ? $row['city'] : '';
+        $state = is_string($row['state'] ?? null) ? $row['state'] : '';
+
         if ($_POST['form_csvexport']) {
             echo csvEscape(oeFormatShortDate(substr((string) $row['edate'], 0, 10))) . ',';
```
Comment (kojiromike) [inline interface/reports/patient_list.php:288]: Is this really going in a csv? It looks like an html table. If it's an html table, is `csvEscape` appropriate? If it's a csv, surely there's a better way to construct a csv than one field at a time. Is there not already an issue for this? If not, create one. This code probably needs some research.

Comment (kojiromike) [reply interface/reports/patient_list.php:288]: The surrounding conditional `if ($_POST['form_csvexport'])` is what distinguishes the two paths: the CSV branch uses `csvEscape()` and the HTML-table branch is the `else` below. The field-by-field construction is definitely worth refactoring though — there's already an existing tracking issue for this exact pattern across the codebase: openemr/openemr#10818 ("refactor(csv): Update various CSV generation paths to use built-in or library tooling"). It specifically calls out the `csvEscape` callers and mentions migrating to `SpreadSheetService` / `fputcsv`.

Out of scope for this literal-string cleanup PR, but the migration is tracked.
